neuro_san/service/agent_server.py

```diff
@@ -106,48 +110,82 @@ def setup_tool_factory_provider(self):
         for agent_name, tool_registry in self.tool_registries.items():
             tool_factory_provider.add_agent_tool_registry(agent_name, tool_registry)
 
+    def agent_added(self, agent_name: str):
+        """
+        Agent is being added to the service.
+        :param agent_name: name of an agent
+        """
+        tool_factory_provider: ServiceToolFactoryProvider =\
+            ServiceToolFactoryProvider.get_instance()
+        service = AgentService(self.server_lifetime, self.security_cfg,
+                               agent_name,
+                               tool_factory_provider.get_agent_tool_factory_provider(agent_name),
+                               self.server_logging)
+        self.services.append(service)
+        servicer_to_server = AgentServicerToServer(service)
+        agent_rpc_handlers = servicer_to_server.build_rpc_handlers()
+        agent_service_name: str = AgentServiceStub.prepare_service_name(agent_name)
+        service_router: DynamicAgentRouter = DynamicAgentRouter().get_instance()
+        service_router.add_service(agent_service_name, agent_rpc_handlers)
+
+    def agent_modified(self, agent_name: str):
+        """
+        Agent is being modified in the service scope.
+        :param agent_name: name of an agent
+        """
+        # Endpoints configuration has not changed,
+        # so nothing to do here, actually.
+        _ = agent_name
```

Comment:
Worth a comment as to how/where the modified agent file is handled.
Are we re-reading a hocon file on every request?
If so that could effect performance adversely (though going there is probably not for this PR), especially for simple requests that should take no time like function() and connectivity()

Reply:
Right now, modified agent is not handled anywhere yet - that's for new feature development.
On this level, we requite already built(compiled) AgentToolFactory, where it comes from is not a concern here.
One of possible options - registry directory watchdog will do this building before calling
ServiceToolFactoryProvider.add_agent_tool_registry()